### PR TITLE
Updating Packer and Script files

### PR DESCRIPTION
**Pull request by kishorekumar-anchala**
# Description

- Updated packer file for ubuntu24.04
- modified dotnetcore.sh file
- added source path

## Check list
- [ 1 ] Related issue / work item is attached
- [ ] Tests are written (if applicable)
- [ ] Documentation is updated (if applicable)
- [ ] Changes are tested and related VM images are successfully generated

### CODE REVIEW

File: images/ubuntu/scripts/build/install-dotnetcore-sdk.sh [hunk 1]
```diff
@@ -63,13 +63,15 @@ apt-get update
...
-    fi
+    releases=$(download_with_retry "$release_url")
+
+    # Extract SDK versions from the releases metadata
+    sdk_versions=($(echo "$releases" | jq -r '.releases[].sdk.version | select(contains("preview") or contains("rc") | not)'))
+    sdks+=("${sdk_versions[@]}")
+
+    # Extract additional SDK versions from the releases metadata
+    additional_sdks=($(echo "$releases" | jq -r '.releases[].sdks[]?.version | select(contains("preview") or contains("rc") | not)'))
+    sdks+=("${additional_sdks[@]}")
```
Comment (Alexey-Ayupov) [inline images/ubuntu/scripts/build/install-dotnetcore-sdk.sh:74]: Why do we change this?

File: images/ubuntu/templates/ubuntu-24.04.pkr.hcl [hunk 2]
```diff
@@ -188,6 +188,11 @@ build {
     script          = "${path.root}/../scripts/build/configure-apt-mock.sh"
   }
 
+  provisioner "file" {
+    destination = "${var.helper_script_folder}"
+    source      = "${path.root}/../scripts/helpers"
+  }
+
```
Comment (Alexey-Ayupov) [inline images/ubuntu/templates/ubuntu-24.04.pkr.hcl:195]: Why did you move this section? Does this change anything?

File: images/ubuntu/templates/ubuntu-24.04.pkr.hcl [hunk 3]
```diff
@@ -203,10 +208,10 @@ build {
...
   }
 
-  provisioner "file" {
-    destination = "${var.helper_script_folder}"
-    source      = "${path.root}/../scripts/helpers"
-  }
+  // provisioner "file" {
+  //   destination = "${var.helper_script_folder}"
+  //   source      = "${path.root}/../scripts/helpers"
+  // }
```
Comment (Alexey-Ayupov) [inline images/ubuntu/templates/ubuntu-24.04.pkr.hcl:214]: Comments in the code are only needed to explain your actions or for the history. If you want to exclude the code from execution just remove it.

File: images/ubuntu/templates/ubuntu-24.04.pkr.hcl [hunk 4]
```diff
@@ -258,11 +263,11 @@ build {
...
-provisioner "shell" {
-    environment_vars = ["HELPER_SCRIPTS=${var.helper_script_folder}", "INSTALLER_SCRIPT_FOLDER=${var.installer_script_folder}"]
-    execute_command  = "sudo sh -c '{{ .Vars }} {{ .Path }}'"
-    scripts          = ["${path.root}/../scripts/build/install-powershell.sh"]
-  }
+  provisioner "shell" {
+      environment_vars = ["HELPER_SCRIPTS=${var.helper_script_folder}", "INSTALLER_SCRIPT_FOLDER=${var.installer_script_folder}"]
+      execute_command  = "sudo sh -c '{{ .Vars }} {{ .Path }}'"
+      scripts          = ["${path.root}/../scripts/build/install-powershell.sh"]
+    }
```
Comment (Alexey-Ayupov) [inline images/ubuntu/templates/ubuntu-24.04.pkr.hcl:270]: Why we need double spaces in this section? Does this change anything?

File: images/ubuntu/scripts/build/configure-apt.sh [hunk 5]
```diff
@@ -42,6 +42,9 @@ apt-get update
 # Install jq
 apt-get install jq
 
+# Source the helpers for use with the script
+source $HELPER_SCRIPTS/os.sh
```
[the remark this answers is not in the document]
Comment (kishorekumar-anchala) [reply images/ubuntu/scripts/build/configure-apt.sh:46]: Sure, Thank you for your input @Alexey-Ayupov .

File: images/ubuntu/scripts/tests/DotnetSDK.Tests.ps1 [hunk 6]
```diff
@@ -20,11 +20,11 @@ Describe "Dotnet and tools" {
             $dotnet = @{ dotnetVersion = $version }
 
             It "SDK <dotnetVersion> is available" -TestCases $dotnet {
-                $dotnetSDKs | Should -Match "$dotnetVersion.[1-9]*"
+                $dotnetSDKs | Should -Match "$dotnetVersion.[0-9]*"
```
[the remark this answers is not in the document]
Comment (kishorekumar-anchala) [reply images/ubuntu/scripts/tests/DotnetSDK.Tests.ps1:23]: Definitely, i will revert this change.

File: images/ubuntu/templates/ubuntu-24.04.pkr.hcl [hunk 7]
```diff
@@ -2,7 +2,7 @@ packer {
   required_plugins {
     azure = {
       source  = "github.com/hashicorp/azure"
-      version = "1.4.5"
+      version = ">=1.4.5"
```
[the remark this answers is not in the document]
Comment (kishorekumar-anchala) [reply images/ubuntu/templates/ubuntu-24.04.pkr.hcl:5]: Sure.

File: images/ubuntu/scripts/build/install-dotnetcore-sdk.sh (hunk 1 above)
Comment (kishorekumar-anchala) [reply images/ubuntu/scripts/build/install-dotnetcore-sdk.sh:74]: Earlier it is looking and downloading  for only dot net 6.0 , it throws error when building the pipeline, so i thought  to remove previous script snippet and added this to remove those boundaries.

File: images/ubuntu/templates/ubuntu-24.04.pkr.hcl (hunk 2 above)
Comment (kishorekumar-anchala) [reply images/ubuntu/templates/ubuntu-24.04.pkr.hcl:195]: i feel packer will execute  in sequence manner, so before execution of configure-apt.sh , we have to copy the file(os.sh) which are available in HELPER folder in git path to $HELPER_FOLDER

File: images/ubuntu/templates/ubuntu-24.04.pkr.hcl (hunk 4 above)
Comment (kishorekumar-anchala) [reply images/ubuntu/templates/ubuntu-24.04.pkr.hcl:270]: Earlier this snippet has some indentation issue, i forgot to remove those spaces. i will rectify them. Thankyou @Alexey-Ayupov 

will create another PR by taking your inputs.

File: images/ubuntu/templates/ubuntu-24.04.pkr.hcl (hunk 3 above)
Comment (kishorekumar-anchala) [reply images/ubuntu/templates/ubuntu-24.04.pkr.hcl:214]: Sure @Alexey-Ayupov , Thanks for your inputs.
